funsor/pyro/distribution.py

```diff
@@ -67,23 +67,23 @@ def _sample_delta(self, sample_shape):
                 if shape[dim] > 1:
                     sample_inputs[DIM_TO_NAME[dim]] = bint(shape[dim])
         delta = self.funsor_dist.sample(frozenset({"value"}), sample_inputs)
-        if isinstance(delta, Joint):
-            delta, = delta.deltas
-        assert isinstance(delta, Delta)
+        if isinstance(delta, Contraction):
```

Comment:
Add an assertion that `len(delta.terms) == `. This was previously checked by the destructured assignment.